internal/controller/processor.go

```diff
@@ -201,17 +199,26 @@ func (processor *AppConfigurationProviderProcessor) processSecretReferenceRefres
 
 	// Only resolve the secret references that not specified the secret version
 	secretReferencesToSolve := make(map[string]*loader.TargetSecretReference)
+	cachedSecretReferences := make(map[string]*loader.TargetSecretReference)
 	for secretName, reference := range reconcileState.ExistingSecretReferences {
 		for key, uriSegment := range reference.UriSegments {
 			if uriSegment.SecretVersion == "" {
 				if secretReferencesToSolve[secretName] == nil {
 					secretReferencesToSolve[secretName] = &loader.TargetSecretReference{
 						Type:        reference.Type,
-						UriSegments: make(map[string]loader.KeyVaultSecretUriSegment),
+						UriSegments: make(map[string]loader.KeyVaultSecretMetadata),
 					}
 				}
 				secretReferencesToSolve[secretName].UriSegments[key] = uriSegment
 			}
+
```

Comment:
I believe you are trying to do map deepcopy here right?

Reply:
yes, do you mean we need to fave a new `loop` for map deepcopy?

Reply:
Does `map.Clone()` help?

Reply:
Have a try with `maps.Clone()` while it failed to apply to `maps[string]*loader.TargetSecretReference`, since the value type is pointer, after clone the new map points to the same struct.